internal/storage/badger/store_test.go

```diff
@@ -62,7 +62,31 @@ func TestPutAndGet(t *testing.T) {
 		if results, err := store.Get([]byte(key)); err != nil {
 			t.Fatalf("Unable to GET. Key: %s, Error: %v", key, err)
 		} else if string(results[0].Value) != expectedValue {
-			t.Errorf("GET mismatch. Key: %s, Expected Value: %s, Actual Value: %s", key, expectedValue, results[0])
+			t.Errorf("GET mismatch. Key: %s, Expected Value: %s, Actual Value: %s", key, expectedValue, results[0].Value)
+		}
+	}
+}
+
+func TestMultiPutAndGet(t *testing.T) {
+	numKeys := 10
+	items := make([]*storage.KVEntry, numKeys+1)
+	for i := 1; i <= numKeys; i++ {
+		key, value := fmt.Sprintf("MPK%d", i), fmt.Sprintf("VALUEXXXX%d", i)
+		items[i] = kvEntry(key, value)
+	}
+
+	if err := store.Put(items...); err != nil {
```

Comment:
any way to test atomicity here ? like if 1 put fails , assert all puts fail.

Reply:
Well, the underlying layer itself uses `NewWriteBatch` which provides this [atomicity of a single transaction](https://github.com/dgraph-io/badger/blob/master/batch.go)